template/base/src/env/server-env.mjs

```diff
@@ -0,0 +1,50 @@
+// @ts-check
+/**
+ * This file is included in `/next.config.mjs` which ensures the app isn't built with invalid env vars.
+ * It has to be a `.mjs`-file to be imported there.
+ */
+ import { serverEnvSchema, clientEnvSchema } from "./env-schema.mjs";
+
+ const formatErrors = (
+   /** @type {import('zod').ZodFormattedError<Map<string,string>,string>} */
+   errors,
+ ) =>
+   Object.entries(errors)
+     .map(([name, value]) => {
+       if (value && "_errors" in value)
+         return `${name}: ${value._errors.join(", ")}\n`;
+     })
+     .filter(Boolean);
+ 
+ const _serverEnv = serverEnvSchema.safeParse(process.env);
+ const _clientEnv = clientEnvSchema.safeParse(process.env);
```

Comment:
Do we need to validate client here? Can we not just import the env from client instead of the schema?

Validating here and checking status leads to code duplication and error messages needing to be changed in two places.

Reply:
Yea that makes more sense, I'll refactor.